web/pages/[[...slug]].tsx

```diff
@@ -241,11 +242,25 @@ const Route = ({ data: initialData, slug, preview }: RouteProps) => {
   );
 };
 
-export const getServerSideProps: GetServerSideProps = async ({
-  params,
+export const getStaticPaths: GetStaticPaths = async () => {
+  const allSlugsQuery = groq`*[defined(slug.current) && _type == 'route'][].slug.current`;
+  const pages = await client.fetch(allSlugsQuery);
+  const paths = pages.map((slug) => ({
+    params: {
+      slug: [urlJoin(slug, { leadingSlash: false })],
+    },
+  }));
+
+  return { paths, fallback: 'blocking' };
+};
+
+export const getStaticProps: GetStaticProps = async ({
+  params: { slug: s = '/' },
   preview = false,
 }) => {
-  const slug = params?.slug?.[0] || '/';
+  const slug = Array.isArray(s)
+    ? urlJoin.apply(null, [...s, { leadingSlash: false }])
+    : s;
```

Comment:
Part 3 of my confusion, since it looks like the code intends to handle the case where there are multiple slug entries, which I don't think can ever happen currently? But maybe it's just intended to take the one and only `s` entry and remove any leading slashes?

Depending on the intentions, maybe a better name would be `path`, because in the URL `https://foo.test/bar/baz` I think the "slug" in standard parlance is really just the `baz` part.

Reply:
I think you're right in that we only need to support one level of slugs so far. All pages I've seen have been accessible on _/:slug_, and I haven't seen any examples of _/:slug/:nested_. Still, I've tried to not needlessly restrict us here.

I've named it `slug` to be consistent with the field which we're querying Sanity by, I feel like this makes more sense in this particular case, and explains the relation to the Sanity field better than naming it `path`. The way we're currently using this route, only Sanity can specify that a Route is available at `/bar/baz`, so `bar` would never _not_ originate from Sanity.

Reply:
OK, let's keep it.